lib/mongo/uri.rb

```diff
@@ -256,6 +256,19 @@ def initialize(string, options = {})
       parsed_scheme, _, remaining = string.partition(SCHEME_DELIM)
       raise_invalid_error!(INVALID_SCHEME) unless parsed_scheme == scheme
       parse!(remaining)
+
+      # Warn if conflicting insecure TLS options are provided.
+      if @uri_options[:ssl_verify] == false
+        if @uri_options[:tls_allow_invalid_certificates] == false
+          log_warn("tlsInsecure/ssl_verify is set to disable verification, but " +
+                     "tls_allow_invalid_certificates is set to enable it")
```

Comment:
Can this message also include which option takes precedence?

Reply:
If the precedence is not currently defined, I would think it is a good idea to define it.